reference/enterprise-api-ref/cmdb-api.markdown

```diff
@@ -0,0 +1,317 @@
+---
+layout: default
+title: CMDB API
+published: true
+tags: [reference, enterprise, API, CMDB, classes, variables]
+---
+
+Configuration management database API enables you to manage classes and variables for specific hosts.
+
+## List CMDB 
+
+You can see a list of stored hosts-specific configurations
+
+**URI:** https://hub.cfengine.com/cmdb
+
+**Method:** GET
+
+**Parameters:**
+* **fromEpoch** *(integer)*
+  Returns configurations with epoch value greater than set in the filter.
+  Epoch is the sequence number of the latest CMDB change. In every API list request, you
+  might see `most_recent_cmdb_change` property on the meta section, which contains the maximum
+  epoch value among selected items. Optional parameter.
+* **fromTime** *(timestamp)*
+  Include changes performed within interval. Format: `YYYY-mm-dd HH:MM:SS` or `YYYY-mm-dd`
+* **toTime** *(timestamp)*
+  Include changes performed within interval. Format: `YYYY-mm-dd HH:MM:SS` or `YYYY-mm-dd`
+* **skip** *(integer)*
+  Number of results to skip for the processed
+  query. The Mission Portal uses this for pagination. Optional parameter.
+* **limit**  *(integer)*
+  Limit the number of results in the query.
+* **hostContextInclude** *(array)*
+  Includes only results that concern hosts which have all specified CFEngine contexts (class) set. Optional parameter.
+* **hostContextExclude** *(array)*
+  Excludes results that concern hosts which have specified CFEngine context (class) set. Hosts that have at least one of the specified contexts set will be excluded from the results. Optional parameter.
+
+    
+**Example request (curl):**
+ 
+```
+curl -k --user <username>:<password> \
+  -X GET \
+  https://hub.cfengine.com/api/cmdb?epochFrom=2&hostContextInclude[]=ubuntu \
+  -H 'content-type: application/json' 
+```
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "data": {
+        "SHA=fdab725e8fb18aa0ad194477be8a2a1338b4b29f6a8597819af89697e432418f": {
+            "classes": {
+                "My_class": {},
+                "My_class2": {
+                    "comment": "comment helps to understand what this class serves for"
+                }
+            },
+            "variables": {
+                "HubCMDB:My.hostname": {
+                    "value": "host1.cfengine.com",
+                    "comment": "comment"
+                },
+                "Namespace:BundleName.VariableName": {
+                    "value": "myvalue"
+                }
+            }
+        }
+    },
+    "meta": {
+        "count": "1",
+        "page": 1,
+        "timestamp": 1619116399,
+        "total": "1",
+        "most_recent_cmdb_change": "13"
```

Comment:
looking at this here I don't like that "most_recent_cmdb_change" has no obvious link to "epoch". That change would be in code as well but it would make more sense to me to call this "cmdb_epoch".

Reply:
@olehermanse @vpodzime care to comment on the colour of this bike shed?

Reply:
`cmdb_epoch` makes more sense to me given the `fromEpoch` argument.

Reply:
renamed to `cmdb_epoch`